numpy/distutils/fcompiler/gnu.py

```diff
@@ -273,7 +273,12 @@ def version_match(self, version_string):
         if not v or v[0] != 'gfortran':
             return None
         v = v[1]
-        if v >= '4.':
+        try:
+            major_version = int(v.split('.', 1)[0])
+        except ValueError:
+            return v
+
+        if major_version >= 4:
```

Comment:
Might make sense to just use distutils here.
```suggestion
        if distutils.version.LooseVersion(v) >= "4":
```

Reply:
thanks I made this change